src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -27,9 +28,27 @@ def __init__(self, pex_path, interpreter):
   def pex(self):
     return self._pex
 
+  def _pretty_cmdline(self, args):
+    return safe_shlex_join(self._pex.cmdline(args))
+
+  def output(self, args, input=None, binary_mode=False, **kwargs):
```

Comment:
Thanks for adding `binary_mode`!

Could we please rename `input` as this shadows the builtin `input`? Maybe something like `data` or `payload`. `process.communicate()` taking `input` as an argument is not a good role model, which is confusing.

Reply:
I've converted this to `stdin_payload`, and also fixed it in `test_git_hooks.py`. emacs highlights `input` as a builtin, I assumed there was some reason the subprocess module decided to use it (always a bad idea).